### PR TITLE
Cleanup MixedViewport

### PR DESCRIPTION
**Pull request by abarth**
This patch fixes a couple minor bugs and cleans up MixedViewport a bit.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/mixed_viewport.dart [hunk 1]
```diff
@@ -198,7 +202,7 @@ class _MixedViewportElement extends RenderObjectElement<MixedViewport> {
       final int startIndex = _firstVisibleChildIndex;
       int lastIndex = startIndex + _childrenByKey.length - 1;
       Element nextSibling = null;
-      for (int index = lastIndex; index >= startIndex; index -= 1) {
+      for (int index = startIndex; index <= lastIndex; index += 1) {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/mixed_viewport.dart]: this should be a breaking change. If this doesn't make a test fail I'm (a) surprised and (b) disappointed that we don't have a test that checks this.

Comment (abarth) [inline packages/flutter/lib/src/widgets/mixed_viewport.dart]: Yes, this changes this pass to be in-order, similar to how we changed the regular pass to be in-order recently.

**Comment by Hixie** [comment]
LGTM with a test that checks the slot logic.

**Pull request merged**